src/NUnitFramework/framework/Api/DefaultTestAssemblyBuilder.cs

```diff
@@ -99,51 +99,42 @@ private TestSuite Build(Assembly assembly, string assemblyNameOrPath, IDictionar
 
             try
             {
-                if (options.ContainsKey(FrameworkPackageSettings.DefaultTestNamePattern))
-                    TestNameGenerator.DefaultTestNamePattern = options[FrameworkPackageSettings.DefaultTestNamePattern] as string;
-
-                if (options.ContainsKey(FrameworkPackageSettings.WorkDirectory))
-                    TestContext.DefaultWorkDirectory = options[FrameworkPackageSettings.WorkDirectory] as string;
-                else
-                    TestContext.DefaultWorkDirectory = Directory.GetCurrentDirectory();
-
-                if (options.ContainsKey(FrameworkPackageSettings.TestParametersDictionary))
-                {
-                    var testParametersDictionary = options[FrameworkPackageSettings.TestParametersDictionary] as IDictionary<string, string>;
-                    if (testParametersDictionary != null)
-                    {
-                        foreach (var parameter in testParametersDictionary)
-                            TestContext.Parameters.Add(parameter.Key, parameter.Value);
-                    }
-                }
+                options.TryGetValue(FrameworkPackageSettings.DefaultTestNamePattern, out object defaultTestNamePattern);
```

Comment:
I think you should retain the `if` semantics while getting the benefits of single lookup by having:

```csharp
if (options.TryGetValue(FrameworkPackageSettings.DefaultTestNamePattern, out object defaultTestNamePattern))
    TestNameGenerator.DefaultTestNamePattern = defaultTestNamePattern as string;
```

Reply:
Ah thanks! Completely forgot that TryGetValue() returned a bool result...